src/Kestrel.Core/Internal/Http/Http1OutputProducer.cs

```diff
@@ -181,17 +179,15 @@ public void Abort(Exception error)
                     return;
                 }
 
+                _aborted = true;
+                _lifetimeFeature.Abort(error);
+
                 if (!_completed)
                 {
                     _log.ConnectionDisconnect(_connectionId);
                     _completed = true;
-
-                    _outputPipeReader.CancelPendingRead();
-                    _pipeWriter.Complete(error);
+                    _pipeWriter.Complete();
```

Comment:
So we never complete the transport with an error.

Reply:
Correct. The Abort method is now the one true way to fail the connection.

Before completing with an error was one way to tell the transport not to worry about any unflushed data (in fact the transport wouldn't even be able to read any more unconsumed data from output pipe after the pipe was completed with the failure), but now that calling IConnectionLifetimeFeature.Abort(ex) cancels any pending reads from the output pipe which breaks the send loop in both transport, this is no longer necessary.